commit 7bb1f0523f81fd999d1e5301625086143c4c7729
Author: Eliad Peller <eliad@wizery.com>
Date:   Wed Jan 4 13:37:30 2012 +0200

    mac80211: set upasd_queues and max_sp only on sta addition
    
    (this patch is not needed for newer kernel versions)

diff --git a/net/mac80211/cfg.c b/net/mac80211/cfg.c
index 7546b60..a122b8c 100644
--- a/net/mac80211/cfg.c
+++ b/net/mac80211/cfg.c
@@ -694,9 +694,6 @@ static void sta_apply_parameters(struct ieee80211_local *local,
 	}
 	spin_unlock_irqrestore(&sta->flaglock, flags);
 
-	sta->sta.uapsd_queues = params->uapsd_queues;
-	sta->sta.max_sp = params->max_sp;
-
 	/*
 	 * cfg80211 validates this (1-2007) and allows setting the AID
 	 * only when creating a new station entry
@@ -794,6 +791,8 @@ static int ieee80211_add_station(struct wiphy *wiphy, struct net_device *dev,
 		set_sta_flags(sta, WLAN_STA_ASSOC);
 
 	sta_apply_parameters(local, sta, params);
+	sta->sta.uapsd_queues = params->uapsd_queues;
+	sta->sta.max_sp = params->max_sp;
 
 	rate_control_rate_init(sta);
 
